pkg/controller/rds/dbparametergroup/setup.go

```diff
@@ -0,0 +1,150 @@
+			managed.WithLogger(l.WithValues("controller", name)),
+			managed.WithRecorder(event.NewAPIRecorder(mgr.GetEventRecorderFor(name)))))
+}
+
+type custom struct {
+	kube   client.Client
+	client svcsdkapi.RDSAPI
+}
+
+func preObserve(_ context.Context, cr *svcapitypes.DBParameterGroup, obj *svcsdk.DescribeDBParameterGroupsInput) error {
+	obj.DBParameterGroupName = awsclients.String(meta.GetExternalName(cr))
+	return nil
+}
+
+func postObserve(_ context.Context, cr *svcapitypes.DBParameterGroup, obj *svcsdk.DescribeDBParameterGroupsOutput, obs managed.ExternalObservation, err error) (managed.ExternalObservation, error) {
+	if err != nil {
+		return managed.ExternalObservation{}, err
+	}
+	cr.SetConditions(xpv1.Available())
+	return obs, err
+}
+
+// func postCreate(_ context.Context, cr *svcapitypes.DBParameterGroup, obj *svcsdk.CreateDBParameterGroupOutput, cre managed.ExternalCreation, err error) (managed.ExternalCreation, error) {
+// 	if err != nil {
+// 		return managed.ExternalCreation{}, err
+// 	}
+// 	meta.SetExternalName(cr, awsclients.StringValue(obj.DBParameterGroup.DBParameterGroupName))
+// 	return managed.ExternalCreation{ExternalNameAssigned: true}, nil
+// }
+
+func preCreate(_ context.Context, cr *svcapitypes.DBParameterGroup, obj *svcsdk.CreateDBParameterGroupInput) error {
+	obj.DBParameterGroupName = awsclients.String(meta.GetExternalName(cr))
+
+	return nil
+}
+
+func preUpdate(_ context.Context, cr *svcapitypes.DBParameterGroup, obj *svcsdk.ModifyDBParameterGroupInput) error {
+	obj.DBParameterGroupName = awsclients.String(meta.GetExternalName(cr))
+	obj.Parameters = make([]*rds.Parameter, len(cr.Spec.ForProvider.Parameters))
+
+	for i, v := range cr.Spec.ForProvider.Parameters {
+		// check if mandatory parameters are set (ApplyMethod, ParameterName, ParameterValue)
+		if (v.ApplyMethod == nil) || (v.ParameterName == nil) || (v.ParameterValue == nil) {
+			return errors.New("ApplyMethod, ParameterName and ParameterValue are mandatory fields and can not be nil")
+		}
+		obj.Parameters[i] = &rds.Parameter{
+			ApplyMethod:    awsclients.String(*v.ApplyMethod),
+			ParameterName:  awsclients.String(*v.ParameterName),
+			ParameterValue: awsclients.String(*v.ParameterValue),
+		}
+	}
+	return nil
+}
+
+func preDelete(_ context.Context, cr *svcapitypes.DBParameterGroup, obj *svcsdk.DeleteDBParameterGroupInput) (bool, error) {
+	obj.DBParameterGroupName = awsclients.String(meta.GetExternalName(cr))
+	return false, nil
+}
+
+func (e *custom) isUpToDate(cr *svcapitypes.DBParameterGroup, obj *svcsdk.DescribeDBParameterGroupsOutput) (bool, error) {
+	// TODO(Dkaykay): We need isUpToDate to have context.
+	ctx := context.TODO()
+	results, err := e.getCurrentDBParameters(ctx, cr)
+
+	// compare CR with currently set Parameters
+	for _, v := range cr.Spec.ForProvider.Parameters {
+		for _, w := range results {
+			if *v.ParameterName == *w.ParameterName {
```

Comment:
You can use `awsclients.StringValue(v.ParameterName)` function and worry about only the value, checking against `""` instead of `nil`.

Reply:
changed following suggestion